File: Cpp/Complex Number Multiplication.cpp
// 0ms, 100.00%; 8.4MB, 75.00%
class Solution
{
public:
    string complexNumberMultiply(const string &a, const string &b)
    {
        int pa = a.find('+'), pb = b.find('+');
        int a1 = stoi(a.substr(0, pa)), a2 = stoi(a.substr(pa + 1, a.size() - pa - 2));
        int b1 = stoi(b.substr(0, pb)), b2 = stoi(b.substr(pb + 1, b.size() - pb - 2));
        int x = a1 * b1 - a2 * b2, y = a1 * b2 + a2 * b1;
        return to_string(x) + '+' + to_string(y) + 'i';
    }
};

File: Python3/Complex Number Multiplication.py
# 36ms, 13.25%; 12.7MB, 100.00%
class Solution:
    def complexNumberMultiply(self, a: str, b: str) -> str:
        a1, a2 = map(int, a.rstrip('i').split('+'))
        b1, b2 = map(int, b.rstrip('i').split('+'))
        x, y = str(a1 * b1 - a2 * b2), str(a1 * b2 + a2 * b1)
        return x + '+' + y + 'i'
